src/standardSchema.ts

```diff
@@ -0,0 +1,179 @@
+  readonly input: Input;
+  readonly output: Output;
+}
+
+export function createStandardSchemaProps<TIn, Output>(
+  schema: AnySchema,
+): StandardSchemaProps<TIn, Output> {
+  /**
+   * Adapts the schema's validate method to the standard schema's validate method.
+   */
+  async function validate(value: unknown): Promise<StandardResult<Output>> {
+    try {
+      const result = await schema.validate(value, {
+        abortEarly: false,
+      });
+
+      return {
+        value: result as Output,
+      };
+    } catch (err) {
+      if (err instanceof ValidationError) {
+        return {
+          issues: issuesFromValidationError(err),
+        };
+      }
+
+      throw err;
+    }
+  }
+
+  return {
+    version: 1,
+    vendor: 'yup',
+    validate,
+  };
+}
+
+function createStandardPath(path: string | undefined): StandardIssue['path'] {
+  if (!path?.length) {
+    return undefined;
+  }
+
+  // Array to store the final path segments
+  const segments: string[] = [];
+  // Buffer for building the current segment
+  let currentSegment = '';
+  // Track if we're inside square brackets (array/property access)
+  let inBrackets = false;
+  // Track if we're inside quotes (for property names with special chars)
+  let inQuotes = false;
+
+  for (let i = 0; i < path.length; i++) {
+    const char = path[i];
+
+    if (char === '[' && !inQuotes) {
+      // When entering brackets, push any accumulated segment after splitting on dots
+      if (currentSegment) {
+        segments.push(...currentSegment.split('.').filter(Boolean));
+        currentSegment = '';
+      }
+      inBrackets = true;
+      continue;
+    }
+
+    if (char === ']' && !inQuotes) {
+      if (currentSegment) {
+        // Handle numeric indices (e.g. arr[0])
+        if (/^\d+$/.test(currentSegment)) {
+          segments.push(currentSegment);
+        } else {
+          // Handle quoted property names (e.g. obj["foo.bar"])
+          segments.push(currentSegment.replace(/^"|"$/g, ''));
+        }
+        currentSegment = '';
+      }
+      inBrackets = false;
+      continue;
+    }
+
+    if (char === '"') {
+      // Toggle quote state for handling quoted property names
+      inQuotes = !inQuotes;
+      continue;
+    }
+
+    if (char === '.' && !inBrackets && !inQuotes) {
+      // On dots outside brackets/quotes, push current segment
+      if (currentSegment) {
+        segments.push(currentSegment);
+        currentSegment = '';
+      }
+      continue;
+    }
+
+    currentSegment += char;
+  }
+
+  // Push any remaining segment after splitting on dots
+  if (currentSegment) {
+    segments.push(...currentSegment.split('.').filter(Boolean));
+  }
+
+  return segments;
+}
+
+function createStandardIssues(
+  error: ValidationError,
+  parentPath?: string,
+): StandardIssue[] {
+  return error.errors.map(
+    (err) =>
+      ({
+        message: err,
+        path: createStandardPath(
+          parentPath ? `${parentPath}.${error.path}` : error.path,
```

Comment:
It looks like this string here is not dependant on the map item `err`, so could it be declared outside of the `.map` for a (very minor) perf improvement?

(And the same goes for the `.flatMap` in `issuesFromValidationError`)